### DISCUSSION TITLE
[App] Simple Kernels

### DISCUSSION THREAD

**Discussion by chhzh123** [cornell-zhang/allo#222] [Projects]
Allo implementations on 1D convolution, partition, dot product, substring matching, etc.

**Comment by sx286** [comment]
Shencheng Xu project progress:
The kernel components for all CCA algorithms have been completed and verified, but the kernel has not yet been optimized and the kernel components have not yet been put together. Attempts are being made to design the filter kernel for FBCCA algorithm.

**Comment by sx286** [comment]
The design of unidirectional and bidirectional zero-order digital filters has been completed, allowing the filters to be filtered on-line and off-line, and attempts are being made to design Chebyshev polynomial kernels based on this to realize Chebyshev filters.

**Comment by chhzh123** [comment]
@sx286 Do you have a Github repo to store your code? Can you also attach the link here?

**Reply by sx286** [reply]
https://github.com/sx286/BCI-Allo Here it is

**Comment by sx286** [comment]
In the past 2 weeks, I successfully assembled the kernel for the CCA algorithm and passed the llvm version of the test using the brain-computer interface dataset, and am currently pushing the algorithm to the backend of the hw emulation for testing.

**Comment by sx286** [comment]
Finished pushing the kernel for the CCA algorithm to the backend and got out the performance metrics such as time needed to compute, accuracy, etc. Attempted to assemble a larger kernel which was able to pass the llvm and csim tests, but could not be pushed into hw mode. Roughly completed the poster needed for the MEng project.

**Comment by sx286** [comment]
Finished the MENG poster. Working on the report right now.

**Comment by am3369** [comment]
Progress for FFT:
Completed a kernel for DFT. Need to still implement the recursion/iterations for FFT using the schedules.

**Comment by am3369** [comment]
Progress update:
Worked through the bugs, but the final output doesn't match the expected output for a specific signal.
